### PR TITLE
API-6921: Vulcan support dateasLong mapping

### PR DESCRIPTION
**Pull request by jabando-va**

### CODE REVIEW

File: src/main/java/gov/va/api/lighthouse/vulcan/mappings/Mappings.java [hunk 1]
```diff
@@ -81,6 +81,17 @@
         new DateMapping.InstantPredicateFactory(DateMapping.defaultGraduatedApproximation()));
   }
 
+  /**
+   * Add a date mapping for Long entity field values using standard "ap" (approximate) date
+   * processing.
+   */
+  public Mappings<EntityT> dateAsLong(String parameterName, String fieldName) {
```
Comment (jblefkove-va) [inline src/main/java/gov/va/api/lighthouse/vulcan/mappings/Mappings.java:88]: A date timestamp could be seconds or milliseconds; the name should specify

File: src/main/java/gov/va/api/lighthouse/vulcan/mappings/DateMapping.java [hunk 2]
```diff
@@ -193,6 +193,53 @@ public Predicate predicate(
       }
     }
   }
+  /*
+  case EQ:
```
Comment (bschofield-va) [inline src/main/java/gov/va/api/lighthouse/vulcan/mappings/DateMapping.java:197]: what's this commented block?

File: src/main/java/gov/va/api/lighthouse/vulcan/mappings/DateMapping.java [hunk 3]
```diff
@@ -193,6 +193,53 @@ public Predicate predicate(
...
+      switch (date.operator()) {
+        case EQ:
+          return criteriaBuilder.and(
+              criteriaBuilder.greaterThanOrEqualTo(field, date.lowerBound().toEpochMilli()),
+              criteriaBuilder.lessThanOrEqualTo(field, date.upperBound().toEpochMilli()));
+        case NE:
+          return criteriaBuilder.or(
+              criteriaBuilder.lessThan(field, date.lowerBound().toEpochMilli()),
+              criteriaBuilder.greaterThan(field, date.upperBound().toEpochMilli()));
+        case GT:
```
Comment (bschofield-va) [inline src/main/java/gov/va/api/lighthouse/vulcan/mappings/DateMapping.java:222]: indicate fall through cases explicitly. This helps convey that you fully intent GT to be treated the same as SA and that you didn't accidentally forget to code something.

File: src/main/java/gov/va/api/lighthouse/vulcan/mappings/DateMapping.java [hunk 4]
```diff
@@ -193,6 +193,53 @@ public Predicate predicate(
...
+        case SA:
+          return criteriaBuilder.greaterThan(field, date.upperBound().toEpochMilli());
+        case LT:
```
Comment (bschofield-va) [inline src/main/java/gov/va/api/lighthouse/vulcan/mappings/DateMapping.java:225]: same

File: src/main/java/gov/va/api/lighthouse/vulcan/mappings/DateMapping.java (hunk 2 above)
Comment (jabando-va) [reply src/main/java/gov/va/api/lighthouse/vulcan/mappings/DateMapping.java:197]: whoops that was when I was testing without specific cases

File: src/test/java/gov/va/api/lighthouse/vulcan/VulcanTest.java [hunk 5]
```diff
@@ -251,6 +251,28 @@ void mappingDateAsInstant() {
         .containsExactlyInAnyOrder(nachos2005, tacos2005, moreNachos2005);
   }
 
+  @SuppressWarnings("SpellCheckingInspection")
+  @Test
+  void mappingDateAsLong() {
```
Comment (bschofield-va) [inline src/test/java/gov/va/api/lighthouse/vulcan/VulcanTest.java:256]: name no longer matches method under test

File: src/main/java/gov/va/api/lighthouse/vulcan/mappings/DateMapping.java [hunk 6]
```diff
@@ -193,6 +193,53 @@ public Predicate predicate(
...
+        case EB:
+          return criteriaBuilder.lessThan(field, date.lowerBound().toEpochMilli());
+        case GE:
+          return criteriaBuilder.greaterThanOrEqualTo(field, date.lowerBound().toEpochMilli());
+        case LE:
+          return criteriaBuilder.lessThanOrEqualTo(field, date.upperBound().toEpochMilli());
+        case AP:
+          return criteriaBuilder.and(
+              criteriaBuilder.greaterThanOrEqualTo(
+                  field, approximation.expandLowerBound(date).toEpochMilli()),
```
Comment (bschofield-va) [inline src/main/java/gov/va/api/lighthouse/vulcan/mappings/DateMapping.java:235]: use the accessor method here to match the style of the other predicate factory

**Review by bschofield-va — APPROVED** [review]

**Comment by bschofield-va** [comment]
Before this is merged... you have tested this with intended usage in DQ?

File: src/test/java/gov/va/api/lighthouse/vulcan/fugazi/FugaziController.java [hunk 7]
```diff
@@ -74,6 +74,7 @@ private FugaziDto asFoo(FugaziEntity e) {
                 .value("xmillis", "millis", v -> Instant.parse(v).toEpochMilli())
                 .values("nameAndFood", this::nameFoodValues)
                 .dateAsInstant("xdate", "date")
+                .dateAsLongMilliseconds("ydate", "millis")
```
Comment (jhulbert-va) [inline src/test/java/gov/va/api/lighthouse/vulcan/fugazi/FugaziController.java:77]: Adding this line means there should be tests added to `VulcanTest`.

**Pull request merged**